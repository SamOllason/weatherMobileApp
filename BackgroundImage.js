import React from 'react'
import { View, Image, StyleSheet, Dimensions} from 'react-native'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        position: 'absolute',
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        width: Dimensions.get('window').width,
        height: Dimensions.get('window').height
    },
    image: {
        flex: 1,
        resizeMode: 'stretch',
    }
});

const BackgroundImage = ({currentWeather}) => {

    const getImage = (weather) => {
        let image = require('./images/clearSky.jpg');

        if(weather) {
            if (weather.toLowerCase() === 'clouds') {
                image = require('./images/cloudy2.jpg');
            }

            if (weather.toLowerCase() === 'rain' || weather.toLowerCase() === 'drizzle') {
                image = require('./images/rain.jpg');
            }

            if (weather.toLowerCase() === 'snow') {
                image = require('./images/snow.jpg');
            }

            if (weather.toLowerCase() === 'clear') {
                image = require('./images/clearSky.jpg');
            }

            if (weather.toLowerCase() === 'mist') {
                image = require('./images/mist.jpg');
            }

            if (weather.toLowerCase() === 'haze') {
                image = require('./images/haze.jpg');
            }

            if (weather.toLowerCase() === 'fog') {
                image = require('./images/fog.jpg');
            }

            if (weather.toLowerCase() === 'dust') {
                image = require('./images/dust.jpg');
            }
        }

        return image;
    };

    const source = getImage(currentWeather);

    return (
            <View style={styles.container} >
                <Image
                    style={styles.image}
                    source={source}
                />
            </View>
        );
};

export default BackgroundImage;
