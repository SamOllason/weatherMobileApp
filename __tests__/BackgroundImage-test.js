import 'react-native';
import React from 'react';
import BackgroundImage from '../BackgroundImage';
import { shallow } from 'enzyme';

// Using Jest + Enzyme
describe('<BackgroundImage />', () => {
    it('renders correctly', () => {
        expect.assertions(1);

        const currentWeatherMock = "cloudy";

        expect(shallow(<BackgroundImage currentWeather={currentWeatherMock}/>)).toMatchSnapshot();
    });

    it('renders one image when weather is rain', () => {
        expect.assertions(1);

        const currentWeatherMock = "rain";
        const wrapper = shallow(<BackgroundImage currentWeather={currentWeatherMock}/>);

        expect(wrapper.find('Image')).toHaveLength(1);
    });

    it('still renders an image when unknown weather passed as props ', () => {
        expect.assertions(1);

        const currentWeatherMock = "unknownweather";
        const wrapper = shallow(<BackgroundImage currentWeather={currentWeatherMock}/>);

        expect(wrapper.find('Image')).toHaveLength(1);
    });

    it('still renders an image when null weather passed as props ', () => {
        expect.assertions(1);

        const currentWeatherMock = null;
        const wrapper = shallow(<BackgroundImage currentWeather={currentWeatherMock}/>);

        expect(wrapper.find('Image')).toHaveLength(1);
    });
});
