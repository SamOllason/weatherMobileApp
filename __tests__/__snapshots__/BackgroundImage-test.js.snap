// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundImage /> renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flex": 1,
      "height": 1334,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": 750,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/clearSky.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "resizeMode": "stretch",
      }
    }
  />
</View>
`;
